wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,160 @@
+  Features like `terminationGracePeriodSeconds` are not sufficient as the termination/migration can
+  take hours if not days.
+- As a user, I want my application to finish all network and storage operations before terminating a
+  pod. This includes closing pod connections, removing pods from endpoints, writing cached writes
+  to the underlying storage and completing storage cleanup routines.
+
+## Deliverables
+
+The WG will coordinate requirement gathering and design, eventually leading to
+KEP(s)s and code associated with the ideas.
+
+Area we expect to explore:
+
+- An API to express node drain/maintenance.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4212.
+- An API to solve the problems wrt the API-initiated Eviction API and PDBs.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4563.
+- An API/mechanism to gracefully terminate pods during a node shutdown.
+  Graceful node shutdown feature tracked in https://github.com/kubernetes/enhancements/issues/2000.
+- An API to deschedule pods that use DRA devices.
+  DRA: device taints and tolerations feature tracked in https://github.com/kubernetes/enhancements/issues/5055.
+- An API to remove pods from endpoints before they terminate.
+  Currently tracked in https://docs.google.com/document/d/1t25jgO_-LRHhjRXf4KJ5xY_t8BZYdapv7MDAxVGY6R8/edit?tab=t.0#heading=h.i4lwa7rdng7y.
+- Introduce enhancements across multiple Kubernetes SIGs to add support for the new APIs to solve
+  wide range of issue.
+
+We expect to provide reference implementations of the new APIs including but not limited to
+controllers, API validation, integration with existing core components and extension points for the
+ecosystem. This should be accompanied by E2E / Conformance tests.
+
+## Relevant Projects
+
+This is a list of known projects that solve similar problems in the ecosystem or would benefit from
+the efforts of this WG:
+
+- https://github.com/aws/aws-node-termination-handler
+- https://github.com/foriequal0/pod-graceful-drain
+- https://github.com/kubereboot/kured
+- https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
+- https://github.com/kubernetes-sigs/karpenter
+- https://github.com/kubevirt/kubevirt
+- https://github.com/medik8s/node-maintenance-operator
+- https://github.com/Mellanox/maintenance-operator
+- https://github.com/openshift/machine-config-operator
+- https://github.com/planetlabs/draino
+- https://github.com/strimzi/drain-cleaner
+
+There are also internal custom solutions that companies use.
+
+## Roles and Organization Management
+
+This WG adheres to the Roles and Organization Management outlined in [wg-governance]
+and opts-in to updates and modifications to [wg-governance].
+
+[wg-governance]: /committee-steering/governance/wg-governance.md
+
+## Timelines and Disbanding
+
+The working group will disband when the KEPs we create are completed. We will
```

Comment:
```suggestion
The working group will disband once the core APIs defined in the KEPs have reached a stable state (GA) and ongoing maintenance ownership is established within the relevant SIGs. We will
```

Reply:
Thanks for the suggestions! Hope it is clear now.